waiter/src/waiter/reporter.clj

```diff
@@ -87,15 +100,53 @@
                s/Any s/Any}
               (merge {:pickled? true} config)))
 
+(defn- make-reporting-name
+  "Concatenates elements to form a dotted name for reporting, eliding any null values or empty strings."
+  [prefix & args]
+  (MetricRegistry/name prefix (into-array String args)))
+
+(let [df (DecimalFormat. "#.####################")]
+  (defn- format-value
+    "Convert numbers to string to send to graphite server"
+    [value]
+    (if (number? value)
+      (.format df value)
+      (str value))))
+
+(defn- report-to-graphite-helper
```

Comment:
A log summarizing how many entries were sent to graphite in a log would be useful for debugging later.